core/arch/arm/plat-hikey/main.c

```diff
@@ -201,10 +200,10 @@ static TEE_Result peripherals_init(void)
 	 * peripherals will not work either (during bootloader stage)
 	 * until linux is booted.
 	 */
-	data = read8(pmussi_base + (PMUSSI_LDO21_REG_ADJ << 2));
-	data = (data & 0xf8) | 0x3;
-	write8(data, pmussi_base + (PMUSSI_LDO21_REG_ADJ << 2));
-	write8(1 << 4, enable_ldo17_22);
+	data = read8(pmussi_base + PMUSSI_LDO21_REG_ADJ);
+	data = (data & PMUSSI_LDO21_REG_MASK) | PMUSSI_LDO21_REG_ENA;
```

Comment:
The masking logic provided is a bit different here, but ok. I'll change the `PMUSSI_LDO21_REG_MASK` value to fit `io_mask8()`.